expyfun/stimuli/_tracker.py

```diff
@@ -738,19 +770,43 @@ def __init__(self, trackers, max_lag=1, rand=None):
         self._response_history = np.array([], dtype=int)
         self._x_history = np.array([], dtype=float)
 
-    def __getitem__(self, key):
-        return self._trackers[key]
+        self._dealer_id = id(self)
+        self._callback('dealer_identity', json.dumps(dict(
+            dealer_id=self._dealer_id)))
+
+        self._callback('dealer_%i_init' % self._dealer_id, json.dumps(dict(
+            trackers=[s._tracker_id for s in self._trackers.ravel()],
+            shape=self._shape,
+            max_lag=self._max_lag,
+            pace_rule=self._pace_rule)))
 
     def __iter__(self):
-        self._index = 0
         return self
 
     def next(self):
-        if self._index == self._trackers.size:
+        """Selects the tracker from which the next trial should be run
+
+        Returns
+        -------
+        subscripts : list-like
+            The position of the selected tracker.
+        x_current : float
+            The level of the selected tracker.
+        """
+        if self.stopped:
             raise(StopIteration)
-        t = self._trackers.flat[self._index]
-        self._index += 1
-        return t
+        if not self._trial_complete:
+            # Chose a new tracker before responding, so record non-response
+            self._response_history = np.append(self._response_history,
+                                               np.nan)
+        self._trial_complete = False
+        self._current_tracker = self._pick()
+        self._tracker_history = np.append(self._tracker_history,
+                                          self._current_tracker)
+        ss = np.unravel_index(self._current_tracker, self.shape)
+        level = self._trackers.flat[self._current_tracker].x_current
```

Comment:
@Eric89GXL This is why `TrackerBinom` needs to have an `x_current` property even if you're using `np.nan`--but are you saying that we should remove the flag for it in `TrackerBinom`?? I'd be hesitant to do so as including it is a nice way to record what level you did the testing at.